internal/common/filespec.go

```diff
@@ -208,18 +213,61 @@ func (this *FileSpec) GetPath() string {
 
 // Copy the target of this FileSpec in the specified location.
 // If the FileSpec has a dest, then that will be the name of the resultant dirent within destDir.
-// If the filespec is a path, then copy a dirent.
+// If the filespec is a path, then copy all matching dirents.
 // If the filespec is a git repo, then ensure it is cloned/updated.
 // Empty and Nil FileSpecs are no-ops.
 // |onlyContents| applies to paths that are dirs and insists that only the contents of dir
 // and not the base dir itself is copied.
+// |baseDir| provides the relative base.
 func (this *FileSpec) CopyTarget(baseDir string, destDir string, onlyContents bool) error {
 	switch this.Type {
 	case FILESPEC_TYPE_EMPTY, FILESPEC_TYPE_NIL:
 		// no-op.
 		return nil
 	case FILESPEC_TYPE_PATH:
-		return this.copyPath(baseDir, destDir, onlyContents)
+		if !filepath.IsAbs(this.Path) && (baseDir != "") {
+			this.Path = filepath.Join(baseDir, this.Path)
```

Comment:
!
Potentially dangerous thing here.
You are joining the baseDir (which can change on every call to CopyTarget()) to this.Path (which lives forever with the filespec) and putting the result in this.Path.
This can create really tough bugs where the order a FileSpec is copied changes later invocations.